python/sdss_access/path/path.py

```diff
@@ -195,64 +246,18 @@ def extract(self, name, example):
         assert name in self.lookup_names(), '{0} must be a valid template name'.format(name)
         template = self.templates[name]
 
-        # expand the environment variable
+        # expand the environment variables
         template = os.path.expandvars(template)
 
-        # handle special functions; perform a drop in replacement
-        if re.match('%spectrodir', template):
```

Comment:
This section of code, lines 201-209 I put in to handle the special function defined in paths using the `%function_name` syntax.  These are functions that get called and replace the `%function_name` with some new path definition subset, which may include the use of keywords.  I wrote this to ensure that `extract` properly returned any keyword arguments used in those functions.  For example, the `plateLines` path defintion.  Does your new `_extract` method handle these cases as well?